mlx/ops.cpp

```diff
@@ -194,6 +194,29 @@ array ones_like(const array& a, StreamOrDevice s /* = {} */) {
   return ones(a.shape(), a.dtype(), to_stream(s));
 }
 
+array eye(int n, int m, int k, Dtype dtype, StreamOrDevice s /* = {} */) {
+  if (n <= 0 || m <= 0) {
+    throw std::invalid_argument("N and M must be positive integers.");
+  }
+  std::vector<int> shape = {n, m};
+  array result = zeros(shape, dtype, s);
+  if (k >= m || -k >= n) {
+    return result;
+  }
+  int start = k >= 0 ? k : 0;
+  int end = k <= 0 ? n + k : m;
+  int length = std::min(n, m) - std::abs(k);
+
+  array diag_indices = arange(start, end * m + start, m + 1, dtype, s);
+  array ones = full({length}, 1, dtype, s);
+  result = scatter(result, {diag_indices}, ones, 0, s);
```

Comment:
There is an overload for scatter which assumes a single index array: `scatter(result, diag_indices, ones, 0, s)`